data-generation/src/main/scala/UpdateDistributions.scala

```diff
@@ -0,0 +1,183 @@
+   * @return A Ordering function
+   * */
+  private def getVertexSorting(mode: UpdateDistributionMode): Ordering[(VertexId, Int)] = {
+    mode match {
+      case UpdateDistributionMode.Uniform => (x: (VertexId, Int), y: (VertexId, Int)) => x._1 compareTo y._1
+      case UpdateDistributionMode.PositiveCorrelation => (x: (VertexId, Int), y: (VertexId, Int)) => x._2 compareTo y._2
+      case UpdateDistributionMode.NegativeCorrelation => (x: (VertexId, Int), y: (VertexId, Int)) => y._2 compareTo x._2
+    }
+  }
+
+  /** Add an update count to all edges
+   *
+   * @param sc           A SparkContext
+   * @param graph        A graph where vertices have a update count as a Int
+   * @param mode         Mode for distributing the weights based on vertex degree
+   * @param distribution Some probability distribution
+   * @param mu           Expected value
+   * @param sigma        Standard deviation
+   * @return */
+  def addEdgeUpdateDistribution[VD, ED: ClassTag](sc: SparkContext, graph: Graph[Int, ED], mode: UpdateDistributionMode, distribution: DistributionType, mu: Int, sigma: Double): Graph[Int, Int] = {
+    val vertexUpdateHashMap = graph
+      .vertices
+      .collect()
+      .toMap
+
+    val graphWithDegree = graph
+      .mapEdges(edge => getVertexUpdateSum[ED](edge, vertexUpdateHashMap))
+      .edges
+      .collect()
+      .sortBy(edge => edge)(getEdgeSorting(mode))
+
+    val updates = graph
+      .mapEdges(_ => getDistributionDraw(distribution, mu, sigma))
+      .edges
+      .collect()
+      .sortBy(edge => edge.attr)
+      .map(edge => edge.attr)
+
+    val rdd = sc.parallelize(
+      graphWithDegree
+        .zip(updates)
+        .map(edgeTuple => Edge(edgeTuple._1.srcId, edgeTuple._1.dstId, edgeTuple._2))
+    )
+    Graph[Int, Int](graph.vertices, rdd)
+  }
+
+  /**
+   *
+   * @param edge
+   * @param hashMap Hash map with vertices as keys and update counts as values
+   * @return The sum of updates of the edge's connected vertices
+   */
+  private def getVertexUpdateSum[ED](edge: Edge[ED], hashMap: Map[VertexId, Int]): Int = {
+    hashMap.getOrElse(edge.srcId, 0) + hashMap.getOrElse(edge.dstId, 0)
+  }
+
+  private def getEdgeSorting(mode: UpdateDistributionMode): Ordering[Edge[Int]] = {
+    mode match {
+      case UpdateDistributionMode.Uniform => (x: Edge[_], y: Edge[_]) => x.hashCode() compareTo y.hashCode()
+      case UpdateDistributionMode.PositiveCorrelation => (x: Edge[Int], y: Edge[Int]) => x.attr compareTo y.attr
+      case UpdateDistributionMode.NegativeCorrelation => (x: Edge[Int], y: Edge[Int]) => y.attr compareTo x.attr
+    }
+  }
+
+  /** Draw a number from a probability distribution
+   *
+   * @param distribution The type of probability distribution
+   * @param mu           Expected value
+   * @param sigma        Standard deviation
+   * @return A value in the distribution
+   *         TODO: Make the input parameter agnostic - not only mu and sigma
```

Comment:
Til denne kommentaren, hvis parameterene var bakt inn i Enum/Case classen så kan det gjøre ting litt enklere

Reply:
Se under for eksempel på hvordan det kunne sett ut